indexing-service/src/main/java/org/apache/druid/indexing/common/task/Task.java

```diff
@@ -331,4 +333,10 @@ static TaskInfo<TaskIdentifier, TaskStatus> toTaskIdentifierInfo(TaskInfo<Task,
         taskInfo.getTask().getMetadata()
     );
   }
+
+  @Nullable
+  default List<String> getLookupsToLoad()
```

Comment:
Please add a javadoc for this.

Reply:
https://github.com/apache/druid/pull/16328#discussion_r1579074507

Reply:
We can use a wrapper as suggested here https://github.com/apache/druid/pull/16328#issuecomment-2075054859

But I am not sure I follow why it would be difficult to pass a nullable parameter via context (especially since null is the default we want).

Reply:
> Keys with null will be difficult to pass via context. Can we use a pojo instead here ?

@cryptoe I'm not entirely sure I follow. I was thinking of passing another key-value pair in `context` which IIUC is an existing task param.
```
context: {
    // existing fields,
    "lookupsToLoad": ["name1", "name2"] // or null
}
```

And the `getLookupsToLoad()` method can parse `context["lookupsToLoad"]` and handle null cases properly. It can also return null if `context` doesn't have `lookupsToLoad` key.

I don't have the end-to-end implementation ready, but this is what I had in my mind currently. Thoughts?

Reply:
@cryptoe @kfaraz Have updated the logic to use a POJO instead of List<String>.